src/repo-query/pages/chat.tsx

```diff
@@ -0,0 +1,313 @@
+import { useEffect, useState } from "react";
+import { REPO_QUERY_QUERY_ENDPOINT } from "../../constants";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import { darcula } from "react-syntax-highlighter/dist/esm/styles/prism";
+
+interface ChatMessage {
+    message: string;
+    isUser: boolean;
+}
+
+export const Chat = ({ ownerName, repoName }: { ownerName: string, repoName: string }) => {
+    // chat ui with bubbles and input
+    const [chatHistory, setChatHistory] = useState<ChatMessage[]>([]);
+    const [sendEnabled, setSendEnabled] = useState(true);
+    const [botThinking, setBotThinking] = useState(false);
+    const [statusMessage, setStatusMessage] = useState("Thinking...🧠");
+
+    const processChunk = (chunk: string) => {
+        const chunkLines = chunk.split("\n");
+        const eventLine = chunkLines[0];
```

Comment:
refactor (nono-blocking): Prefer array destructuring.

```suggestion
        const [eventLine] = chunkLines;
```